ds-collector/ds-collector

```diff
@@ -722,6 +705,81 @@ collect_dse_solr_cores() {
     fi
 }
 
+discover_jmx() {
+ tmp=""
+
+  # XXX from Cassandra 4.0 it is possible to have multiple process per ip, how can we handle this?
+  server_cmdline="$(ps -aef|grep -e 'org.apache.cassandra.service.CassandraDaemon' -e 'com.datastax.bdp.DseModule'|grep java)"
```

Comment:
(trivial)
I'd done something similar in some of my refactoring to get a line-separated list of command-line arguments:
```node_connect "ps -aeo "%a"|grep -E 'org.apache.cassandra.service.CassandraDaemon|com.datastax.bdp.DseModule' | grep -v grep" | tr -d '\r' | head -1 | tr ' ' '\n' > .serverProcess.out 2>&1```

And then with the local file back, I could interrogate lots of things about the Java command line. Both techniques have the unfortunate expectation that the full Java command line is available from the `ps` listing (for example, `/proc` is mounted with `hidepid` option).

Reply:
took some of this.